### PR TITLE
BUG: Fix additional segmentation faults with empty Composite - For Release

### PR DESCRIPTION
**Pull request by blowekamp**
When the CompositeTransform contains no transform is should behave
like an identity transform. This patches address loop iteration issues
that would cause a segmentation fault due to invalid indexing of
transforms when there composite was empty.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
The patch should be merged into the release branch after the nightly results of #1851 being merged are verified.

File: Modules/Core/Transform/test/itkCompositeTransformTest.cxx [hunk 1]
```diff
@@ -127,6 +127,33 @@ itkCompositeTransformTest(int, char *[])
...
+  ITK_TEST_EXPECT_EQUAL(compositeTransform->GetParameters().Size(), 0u);
+  ITK_TEST_EXPECT_EQUAL(compositeTransform->GetFixedParameters().Size(), 0u);
+
+  {
+    CompositeType::InputPointType  inputPoint({ 1.1, 2.2 });
+    CompositeType::InputVectorType inputVector;
+    inputVector[0] = 9.1;
+    inputVector[1] = 8.2;
+
+    if (!testPoint(inputPoint, compositeTransform->TransformPoint(inputPoint)))
```
Comment (DVigneault) [inline Modules/Core/Transform/test/itkCompositeTransformTest.cxx:139]: LGTM!  Could consider using `ITK_TEST_EXPECT_TRUE` here for brevity.

Comment (blowekamp) [reply Modules/Core/Transform/test/itkCompositeTransformTest.cxx:141]: Thanks for the tip. When fixing a bug, and writing a test corresponding to the issue, it is always a struggle not to get carried away with other code improvements. Here I just use the same methodology in the rest of the file for consistency.

**Comment by blowekamp** [comment]
CI failures are not related to this PR.

**Pull request merged**